jschon/vocabulary/core.py

```diff
@@ -78,6 +80,14 @@ def __init__(self, parentschema: JSONSchema, value: str):
 
         parentschema.uri = uri
 
+    def _get_uri_validate_kwargs(self):
+        return {'allow_non_empty_fragment': False}
+
```

Comment:
This can be inlined - as per comment for `IdKeyword_Next`, and with reference to previous discussion around the keyword class hierarchy, keyword implementations should be self-contained.

Reply:
Got it.  This was a bit on auto-pilot as I worked for years on a testing library that managed product versioning through inheritance.